sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java

```diff
@@ -361,9 +361,8 @@ public AccessToken authenticateWithAzureCli(TokenRequestContext request) {
         String tenant = IdentityUtil.resolveTenantId(tenantId, request, options);
         ValidationUtil.validateTenantIdCharacterRange(tenant, LOGGER);
 
-        if (!CoreUtils.isNullOrEmpty(tenant)) {
+        if (!CoreUtils.isNullOrEmpty(tenant) && !tenant.equals(IdentityUtil.DEFAULT_TENANT)) {
```

Comment:
In .NET, I believe we unconditionally use the tenantID. I'm just curious why this logic is needed.

Reply:
`organizations` is not a valid tenant for getting a token from a service principal. A service principal should always be bound to a given tenant. The customer was not passing this through - we made a change a while go that introduced this bug and caused the issue. After researching and talking to @jiasli it was determined that we should never pass this to azcli so we put the check in.